### ISSUE TITLE
Elixir track changes

### ISSUE THREAD

**Issue by neenjaw** [exercism/exercism#4935]
Related to #4699 -- exercise/elixir has a large track change planned to migrate the track from single-file-based to mix-project style. See this pull request here: [https://github.com/exercism/elixir/pull/480](https://github.com/exercism/elixir/pull/480)

Working with @NobbZ, @Cohen-Carlisle @devonestes, @Cohen-Carlisle and I believe we are almost at a reasonable point to migrate.  But in order to make sure that it is complete, we are looking for some feedback on the documentation for the website copy and what we need to consider to collaborate the changes. 

@iHiD, could you point us in the right direction? (I know you're on vacation, just tagging this for you when you return).

**Comment by iHiD** [comment]
Hello :) Thanks for tagging me!

You mention `website-copy` - Do you actually mean `website-copy` or the `website` repo itself?

In terms of what is happening on your track, does this set of changes change the config.json file? If not, I don't think anything needs to happen. If so, could you explain what will change please?

**Comment by neenjaw** [comment]
Overall, the big change is to migrate each exercise from a single elixir script file to a mix project file structure. 

So there will be changes for students when they download the exercise, they should receive the whole file structure instead of just 2 files.

There are also documentation changes to guide students on how to download, run, test the new format. 

There is also some change to the Travis CI script (though I'm not sure this affects more then the elixir repo). 

---

I'll need to double check the config.json but maybe @Cohen-Carlisle can best comment on that. 

Basically, we want to know what happens when we merge it? What might this break? Who do we need to notify?

**Comment by iHiD** [comment]
I don't think anything will break unless the `config.json` changes in an unexpected way (e.g. uuids changing for exercises) or git history is rewritten (both of which are big no-nos) Tracks are pretty self-contained other than the config.json.

Do any of you have thoughts on how/why it might break? :)

**Comment by neenjaw** [comment]
No, I don't think there are any breaking changes in the config. I think we were just being cautious with a sweeping track change

**Comment by iHiD** [comment]
OK great. Then I think you'll be fine :) But if you co-ordinate with me on Slack when you do it I can be there in case something goes wrong.

**Comment by Cohen-Carlisle** [comment]
I thought there were corresponding changes in the `website-copy` repo, but perhaps I misunderstood? As for `config.json`, it did not change in this PR.

**Comment by iHiD** [comment]
If the `config.json` hasn't changed then the exercise ids won't have changed so the database won't be affected. And if the slugs of the exercises haven't changed then nothing needs changing in `website-copy`. If the mentor notes will be wrong after this change then they might want updating , but that is less of a critical/blocking issue I think.

It was definitely worth checking anyway :)

**Comment by neenjaw** [comment]
Alright, I think we are set to merge in the next couple days! Thanks again @iHiD!

**Comment by iHiD** [comment]
Exciting!!
